pkg/crypto/webkms/remotecrypto.go

```diff
@@ -373,6 +387,12 @@ func (r *RemoteCrypto) ComputeMAC(data []byte, keyURL interface{}) ([]byte, erro
 		return nil, err
 	}
 
+	if r.opts.ComputeMACCache != nil {
+		if err := r.opts.ComputeMACCache.Set(string(data), macBytes); err != nil {
```

Comment:
The cache key should be both data and keyURL.

You might want to consider hashing.

Reply:
done